File: DataSetSlection/ProductSearch/step2.rb
# 長さNの文字列Sが与えられます。Sに含まれている各文字の出現回数をそれぞれ求め、「a」の出現回数、「b」の出現回数、...、「z」の出現回数を順に半角スペース区切りで出力してください。

n = gets.to_i
s = gets.chomp.split("")

alp = ("a".."z").to_a
ans = []

alp.each do |i|
    count = 0
    s.each do |j|
        if i == j
            count +=1
        end
    end
    ans << count
end

puts ans.join(" ")

File: Python/DataSetSelectionMenu/UnionOfSets/step2.py
# N 個の要素からなる数列Aが与えられます。数列Aは昇順にソートされています。Aの重複した要素を取り除いて昇順に出力してください。

n = int(input())
numbers = set(map(int,input().split())) #setにして重複を削除、intにすることで順序が整う
numbers = list(map(str,numbers)) #結合時の都合により、一度文字列に戻してリスト化
print(" ".join(numbers))